alpha/books/RayTracingInOneWeekend.html

```diff
@@ -882,24 +888,27 @@
 -----------------------------
 First, let’s get ourselves a surface normal so we can shade. This is a vector that is perpendicular
 to the surface at the point of intersection. There are two design decisions to make for normals.
-The first is whether these normals are unit length. That is convenient for shading so I will say
-yes, but I won’t enforce that in the code. This could allow subtle bugs, so be aware this is
-personal preference as are most design decisions like that. For a sphere, the outward normal is in
-the direction of the hit point minus the center:
+The first is whether or not these normals are unit length. This is convenient for shading so I will
+say yes, but we won’t enforce that in the code. This could allow for subtle bugs, so be aware that
+this is a personal preference. Note that many of the design decisions in this book prioritize
+convenience over absolute perfection.
```

Comment:
We are definitely setting normal vectors to unit length, and are depending on that in the code.

The issue here is that if you _don't_ assume unit normals, then any code that needs it will have to always normalize the vectors, leading to cases where a normal vector is normalized multiple times, leading to worse performance. On the other end, the geometric primitives themselves have enough knowledge to generate unit-length normals in many cases. In our sphere code, we divide the normal by the sphere radius: an optimization that avoids the need for a square root. Our new quad code explicitly normalizes the normal vector because it's needed for other operations, so the cost to providing this is effectively zero.

Reply:
Should I take a stab at communicating the above? Seems useful.

Reply:
Sure. I'll have updates in a minute.